### PR TITLE
Fix HA Node ID

### PR DESCRIPTION
**Pull request by AlexanderRMcNeill**
#### PR Checklist
- [x] Chart Version bumped
- [x] CHANGELOG.md updated
- [x] Variables and other changes are documented in the README.md

**What this PR does / why we need it**:
This PR prevents the HA node ID exceeding 40 characters when a release name is over 16 characters. Exceeding a 40 character HA node ID results in errors from Nuget acquiring an HA lock. 

This PR fixes this by using the pod UID for the HA node ID which is always 36 characters instead of using the pod name which is variable based on the release name.

**Which issue this PR fixes** *
This doesn't fix the following issue but avoids it:
https://www.jfrog.com/jira/browse/RTFACT-18851

**Special notes for your reviewer**:

### CODE REVIEW

File: stable/artifactory-ha/templates/artifactory-node-statefulset.yaml [hunk 1]
```diff
@@ -270,7 +270,7 @@ spec:
         - name: HA_NODE_ID
           valueFrom:
             fieldRef:
-              fieldPath: metadata.name
+              fieldPath: metadata.uid
```
Comment (AlexanderRMcNeill) [inline stable/artifactory-ha/templates/artifactory-node-statefulset.yaml]: I'm not sure if there is an issue with the HA Node ID changing every time a new pod comes up. 

I initially tried to trim the release name off the pod name but had no luck due to the pod name not being available when the helm renders the templates.

Comment (eldada) [reply stable/artifactory-ha/templates/artifactory-node-statefulset.yaml]: It is a problem as this id is used to register the node in the cluster and each node consumes a single license. After each pod redeployment, you'll get a new node registered in the cluster, which will show up on the list of instances and will not get a license. 
Need to look at maybe using a template for this.

Comment (AlexanderRMcNeill) [reply stable/artifactory-ha/templates/artifactory-node-statefulset.yaml]: Looks like sprig has a uuid function (https://masterminds.github.io/sprig/uuid.html) that might also work. This means you would get a new HA Node ID every time you release which is slightly better.

Comment (eldada) [reply stable/artifactory-ha/templates/artifactory-node-statefulset.yaml]: Still - means every upgrade will force new IDs. We need to persist the IDs. One of the reasons for using a statefulset.

Comment (AlexanderRMcNeill) [reply stable/artifactory-ha/templates/artifactory-node-statefulset.yaml]: Ok, I'll create an issue for this so it is documented for anyone who runs into it while we are trying to find as solution 😃 

This is not a big deal for us now that we know what was causing our nuget issues but had us scratching our heads for awhile.

**Pull request closed without merging**